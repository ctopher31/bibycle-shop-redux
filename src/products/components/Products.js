import React, { Component } from 'react';
import ProductListing from './ProductListing';

class Products extends Component {
  componentDidMount() {
    this.props.loadProducts();
  }

  render() {
    const { items, addItem } = this.props;

    return (
      <main className="content shop">
        <h1 className="page--heading">Shop our Products</h1>
        <section className="product--section">
          {items.length < 1 ? (
            <div>Loading...</div>
          ) : (
            items.map(item => <ProductListing key={item.number} {...{ ...item, addItem }} />)
          )}
        </section>
      </main>
    );
  }
}

export default Products;
