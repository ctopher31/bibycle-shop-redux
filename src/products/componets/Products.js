import React from 'react';
import ProductListing from './ProductListing';

const Products = ({ products, addItem }) => (
  <main className="content shop">
    <h1 className="page--heading">Shop our Products</h1>
    <section className="product--section">
      {products.length < 1 ? (
        <div>Loading...</div>
      ) : (
        products.map(item => <ProductListing key={item.number} {...{ ...item, addItem }} />)
      )}
    </section>
  </main>
);

export default Products;
