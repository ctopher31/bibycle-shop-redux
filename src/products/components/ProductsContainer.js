import { connect } from 'react-redux';
import { addItemAction } from '../sagas';
import Products from './Products';

const mapStateToProps = state => ({ products: state.products });

const mapDispatchToProps = dispatch => ({ addItem: key => dispatch(addItemAction(key)) });

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Products);
