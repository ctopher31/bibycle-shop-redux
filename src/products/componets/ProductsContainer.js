import { connect } from 'react-redux';
import { addItemAction } from '../actions';
import Products from './Products';

const mapStateToProps = state => ({ products: state.products });

const mapDispatchToProps = dispatch => ({ addItem: key => dispatch(addItemAction(key)) });

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Products);
