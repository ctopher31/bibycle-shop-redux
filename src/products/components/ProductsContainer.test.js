import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import ProductsContainer from './ProductsContainer';
import Products from './Products';
import productsReducer from '../reducers';

describe('ProductsContainer', () => {
  it('should render the container component', () => {
    const store = createStore(productsReducer, {
      products: [
        {
          number: '51001',
          name: 'widget',
          price: 35.57,
          salePrice: 32.95,
          onSale: true,
        },
      ],
      addItem: () => 1,
    });

    const wrapper = mount(
      <Provider store={store}>
        <ProductsContainer />
      </Provider>
    );

    expect(wrapper.find(ProductsContainer).length).toEqual(1);
    const container = wrapper.find(ProductsContainer);
    expect(container.find(Products).length).toEqual(1);
    expect(store.getState().content).toMatchSnapshot();
  });
});
