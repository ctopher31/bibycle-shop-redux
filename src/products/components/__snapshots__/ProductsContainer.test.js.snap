// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Products should render the container component 1`] = `undefined`;

exports[`ProductsContainer should render the container component 1`] = `undefined`;
